test/SlowTests/Issues/RavenDB_12040.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Xunit;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_12040 : RavenTestBase
+    {
+        [Fact]
+        public async Task Should_limit_number_of_stored_slow_io_hints()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var database = await GetDatabase(store.Database);
+
+                for (int i = 0; i < SlowWritesDetails.MaxNumberOfWrites + 10; i++)
+                {
+                    database.NotificationCenter
+                        .SlowWrites
+                        .Add($"C:\\Raven\\Indexes\\MyIndex\\{i}.journal", 1, 10);
+
+                    Thread.Sleep(1);
```

Comment:
That said, can we avoid the `Thread.Sleep` (will take 0.51 seconds here) in favor of passing the date directly?

Reply:
That's fine, let's get this merged.